From 21770fa06b1ece531c4b946af56db9cd860810c9 Mon Sep 17 00:00:00 2001
From: Vaibhav Hiremath <hvaibhav@ti.com>
Date: Fri, 16 Jan 2009 15:39:42 +0530
Subject: [PATCH 181/364] DVI 720P and 480P support added

DVI 720P and 480P support added through compile time option
	- CONFIG_480P
	- CONFIG_720P
---
 drivers/video/omap2/Kconfig         |   18 ++++++++++++++++++
 drivers/video/omap2/panel-generic.c |   13 ++++++++++---
 2 files changed, 28 insertions(+), 3 deletions(-)

diff --git a/drivers/video/omap2/Kconfig b/drivers/video/omap2/Kconfig
index ddb1338..7753000 100644
--- a/drivers/video/omap2/Kconfig
+++ b/drivers/video/omap2/Kconfig
@@ -44,6 +44,24 @@ config PANEL_GENERIC
         help
 	  Generic panel driver.
 	  Used for DVI output for Beagle and OMAP3 SDP.
+choice
+	prompt "Resolution"
+	depends on PANEL_GENERIC
+	default RES_480P
+
+config RES_480P
+	bool "Use 480P resolution"
+	help
+	  Select this option if you want to use the 480P (720x480)
+	  resolution on DVI output.
+
+config RES_720P
+	bool "Use 720P resolution"
+	help
+	  Select this option if you want to use the 720P (1280x720)
+	  resolution on DVI output.
+
+endchoice
 
 config PANEL_SHARP_LS037V7DW01
         tristate "Sharp LS037V7DW01 LCD Panel"
diff --git a/drivers/video/omap2/panel-generic.c b/drivers/video/omap2/panel-generic.c
index 5c8fecd..2fdb47e 100644
--- a/drivers/video/omap2/panel-generic.c
+++ b/drivers/video/omap2/panel-generic.c
@@ -64,10 +64,17 @@ static struct omap_panel generic_panel = {
 	.resume		= generic_panel_resume,
 
 	.timings = {
-		/* 640 x 480 @ 60 Hz  Reduced blanking VESA CVT 0.31M3-R */
-		.x_res		= 640,
+		/* 1280 x 720 @ 60 Hz  Reduced blanking VESA CVT 0.31M3-R */
+#if defined (CONFIG_RES_720P)
+		.x_res		= 1280,
+		.y_res		= 720,
+		.pixel_clock	= 74250,
+#elif defined (CONFIG_RES_480P)
+		/* 720 x 480 @ 60 Hz  Reduced blanking VESA CVT 0.31M3-R */
+		.x_res		= 720,
 		.y_res		= 480,
-		.pixel_clock	= 23500,
+		.pixel_clock	= 30000,
+#endif
 		.hfp		= 48,
 		.hsw		= 32,
 		.hbp		= 80,
-- 
1.6.0.3

